Assets/Scripts/SS3D/Systems/Roles/RoleSystem.cs

```diff
@@ -0,0 +1,195 @@
+using FishNet.Object;
+using SS3D.Core.Behaviours;
+using SS3D.Systems.PlayerControl;
+using UnityEngine;
+using SS3D.Systems.Entities.Events;
+using Coimbra.Services.Events;
+using SS3D.Systems.PlayerControl.Events;
+using SS3D.Logging;
+using System.Collections.Generic;
+using System.Linq;
+using SS3D.Systems.Entities;
+using SS3D.Systems.Inventory.Containers;
+using SS3D.Systems.Inventory.Items;
+using SS3D.Core;
+using SS3D.Data.Enums;
+
+namespace SS3D.Systems.Roles
+{
+    public class RoleSystem : NetworkSystem
+    {
+        [SerializeField] private RolesAvailable rolesAvailable;
+        private List<RoleCounter> roleCounters = new List<RoleCounter>();
+        private Dictionary<Soul, RoleData> rolePlayers = new Dictionary<Soul, RoleData>();
+
+        #region Setup
+        protected override void OnStart()
+        {
+            base.OnStart();
+            Setup();
+        }
+
+        [Server]
+        private void Setup()
+        {
+            AddHandle(OnlineSoulsChanged.AddListener(HandleOnlineSoulsChanged));
+            AddHandle(PlayerContainersReady.AddListener(HandlePlayerContainersReady));
```

Comment:
Remove the player containers ready event, make a direct call to it from where it is invoked.

Reply:
No idea what you mean lol²